### ISSUE TITLE
word embeddings for OOV words

### ISSUE THREAD

**Issue by dreamk73** [anandaswarup/phrase_break_prediction_IS16#1]
I enjoyed reading your paper on phrase break prediction and am grateful that you made this code available. However, I am wondering about developing a general phrase break algorithm for any English text, how to deal with training word embeddings for a large dataset and how to get good results when your test set has many OOV words in it. Do you have any ideas about that?

**Comment by anandaswarup** [comment]
Sorry for the very late response. I was away for nearly 2 months for personal reasons, with very limited connectivity.

(1) For training word embeddings on a large dataset, word2vec can be used (Mikolov, Tomas; et al. "Efficient Estimation of Word Representations in Vector Space". arXiv:1301.3781)

(2) Dealing with OOV words: The easiest way to deal with it, is to rank all words in the training set by their frequency of occurrence. Then we retain the top 90% of the words and rewrite the remaining 10% words in the corpus with a special token <UNK>, and perform the training as usual. After training the special <UNK> token also has a vector representation. During test all OOV words are mapped to the vector representing the <UNK> token.

I hope that this brief explanation answers your queries.
